sql/src/test/java/io/crate/planner/consumer/ManyTableConsumerTest.java

```diff
@@ -197,4 +200,58 @@ public void test3TableSortOnWhere() throws Exception {
         assertThat(t1Andt3.toString(), is("join.doc.t1.doc.t3"));
         assertThat(root.right().getQualifiedName().toString(), is("doc.t2"));
     }
+
+    private static final Map<QualifiedName, AnalyzedRelation> sources = ImmutableMap.of(
+        new QualifiedName(T3.T1_INFO.ident().name()), T3.TR_1,
+        new QualifiedName(T3.T2_INFO.ident().name()), T3.TR_2,
+        new QualifiedName(T3.T3_INFO.ident().name()), T3.TR_3,
+        new QualifiedName(T3.T4_INFO.ident().name()), T3.TR_4);
+
+    private static final SqlExpressions expressions = new SqlExpressions(sources);
+
+    @Test
+    public void testReordering4TableSortOnWhereAndJoinConditionsThatDontMatchThePair() throws Exception {
+        List<QualifiedName> relations = ImmutableList.of(
+            T3.T1,
+            T3.T2,
+            T3.T3,
+            T3.T4);
+        List<JoinPair> joinPairs = ImmutableList.of(
+            new JoinPair(T3.T1, T3.T2, JoinType.INNER, expressions.asSymbol("t1.a=t2.b")),
+            new JoinPair(T3.T2, T3.T3, JoinType.INNER, expressions.asSymbol("t2.b=t1.a")),
```

Comment:
yep this is exactly what I wanted to catch the case that the previous code was using a doing a reordering even though it has no more joins